packages/jest-config/src/normalize.ts

```diff
@@ -745,6 +745,52 @@ export default function normalize(
         }
         break;
       }
+      case 'displayName': {
+        const displayName = oldOptions[key] as Config.DisplayName;
+        const isDisplayNameAnObject =
+          typeof displayName === 'object' && !Array.isArray(displayName);
+        if (typeof displayName === 'string') {
+          value = displayName;
+          break;
+        }
+        /**
+         * Ensuring that displayName shape is correct here so that the
+         * reporters can trust the shape of the data
+         * TODO: Normalize "displayName" such that given a config option
+         * {
+         *  "displayName": "Test"
+         * }
+         * becomes
+         * {
+         *   displayName: {
+         *     name: "Test",
+         *     color: "white"
+         *   }
+         * }
+         *
+         * This can't be done now since this will be a breaking change
+         * for custom reporters
+         */
+        if (isDisplayNameAnObject) {
+          const errorMessage =
```

Comment:
Opting to do a custom error here. Didn't want to overcomplicate it. Open for suggestions here